envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -125,6 +141,17 @@ private fun validateDomainFormat(
     }
 }
 
+private fun validateDomainPatternFormat(
+    it: RawDependency
+) {
+    val domainPattern = it.domainPattern.orEmpty()
+    if (domainPattern.startsWith("http://") || domainPattern.startsWith("https://")) {
+        throw NodeMetadataValidationException(
+            "Unsupported format for domainPattern"
```

Comment:
This is an unspecific error message to a very specific error.
```suggestion
            "Unsupported format for domainPattern: domainPattern cannot contain a schema like http:// or https://"
```